clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

```diff
@@ -78,16 +82,52 @@ object GCBackupAndRestore {
     }
   }
 
+
+
+  // control copy speed
+  //  public static final String CONF_LABEL_DIRECT_WRITE = "distcp.direct.write";
+
   // Construct a DistCp command of the form:
   // `hadoop distcp -D<per storage credentials> -f absoluteAddressesTextFilePath dstNamespaceForHadoopFs`
   // example for command format https://docs.lakefs.io/integrations/distcp.html
+  // with distCp options from https://hadoop.apache.org/docs/r3.2.1/hadoop-distcp/DistCp.html#Command_Line_Options
   def constructDistCpCommand(
-      hadoopProps: Array[(String, String)],
-      absoluteAddressesTextFilePath: String,
-      dstNamespaceForHadoopFs: String
-  ): Array[String] = {
+                              hadoopProps: Array[(String, String)],
+                              absoluteAddressesTextFilePath: String,
+                              dstNamespaceForHadoopFs: String,
+                              hc: Configuration,
+                              numObjectsToCopy: Long
+                            ): Array[String] = {
+
+    val distCpLogsPath = hc.get(DistCpConstants.CONF_LABEL_LOG_PATH, dstNamespaceForHadoopFs + "_distCp/logs/")
+    print("distCpLogsPath: " + distCpLogsPath + "\n")
+
+    // Tune distCp options that control the speed of the file-to-copy list building stage
+    val numListstatusThreads = hc.get(DistCpConstants.CONF_LABEL_LISTSTATUS_THREADS, DistCpMaxNumListStatusThreads.toString)
+    print("numListstatusThreads: " + numListstatusThreads + "\n")
+
+    // Tune distCp options that control the speed of the copy stage
+    // https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.4/bk_cloud-data-access/content/distcp-perf-mappers.html
+    val maxMaps = hc.get(DistCpConstants.CONF_LABEL_MAX_MAPS, max(DistCpConstants.DEFAULT_MAPS, numObjectsToCopy / 1000).toString) //TODO: what is a reasonable number of maps
+    print("maxMaps: " + maxMaps + "\n")
+    val mapsBandwidth = hc.get(DistCpConstants.CONF_LABEL_BANDWIDTH_MB, DistCpConstants.DEFAULT_BANDWIDTH_MB.toString)
+    print("mapsBandwidth: " + mapsBandwidth + "\n")
+
     hadoopProps.map((prop) => "-D" + prop._1 + "=" + prop._2) ++
```

Comment:
Consider
```suggestion
    hadoopProps.map((prop) => s"-D${prop._1}=${prop._2}") ++
```
or something like
```suggestion
    hadoopProps.map({ case (k, v) => s"-D$k=$v" }) ++
```